docs/ingress-egress.md

```diff
@@ -0,0 +1,126 @@
+# ARO HCP Infrastructure Ingress and Egress
+
+This document provides an overview of ingress and egress configurations specific to the ARO HCP infrastructure. It is limited to the control plane components and infrastructure of the ARO HCP platform itself and does not address ingress or egress for customer workloads. It covers service clusters, management clusters, and the services deployed within them.
+
+## Ingress
+
+Ingress is documented separately for service and management clusters, as the mechanisms and responsibilities differ: service cluster ingress is based on Istio, while management cluster ingress is provisioned directly by the Hypershift operator.
+
+### Service Cluster Ingress
+
+This section describes how ingress traffic targeting ARO HCP services—such as the RP frontend, and eventually the Admin API and Backplane—is handled on service clusters. A shared Istio ingress gateway is deployed per service cluster to expose these components securely via a dedicated public IP.
+
+```mermaid
+flowchart TD
+    ExternalTraffic[External Traffic] --> AzureLB[Azure Load Balancer<br/>Public IP]
+    AzureLB --> NSG[NSG<br/>ARM, Geneva, SAW IP Tags]
+    NSG --> AKSCluster[AKS Service Cluster]
+    AKSCluster --> IstioGateway[Istio Ingress Gateway<br/>aks-istio-ingress namespace]
+    IstioGateway --> GatewayResource[Gateway Resource<br/>Hostname-based routing]
+    GatewayResource --> RPFrontend[RP Frontend API]
+    GatewayResource --> BackplaneAPI[Backplane API<br/>Future]
+    GatewayResource --> AdminAPI[Admin API<br/>Future]
+
+    KeyVault[Azure Key Vault] --> TLSCerts[TLS Certificates]
+    TLSCerts --> CSIStore[CSI Secret Store]
+    CSIStore --> IstioGateway
+
+    DNSRecords[DNS Records] --> AzureLB
+
+    style AzureLB fill:#e1f5fe
+    style IstioGateway fill:#f3e5f5
+    style RPFrontend fill:#e8f5e8
+    style BackplaneAPI fill:#fff3e0,stroke-dasharray: 5 5
+    style AdminAPI fill:#fff3e0,stroke-dasharray: 5 5
+```
+
+ARO HCP service clusters expose selected control plane services through a shared Istio ingress gateway. This gateway acts as the central entry point for external traffic targeting these services. This ingress gateway is deployed into the `aks-istio-ingress` namespace and is currently responsible only for exposing the RP frontend API. In the future, the same gateway will also serve the Backplane and Admin API, and the structure of the associated Helm chart will be re-evaluated accordingly.
```

Comment:
`and the structure of the associated Helm chart will be re-evaluated accordingly` that is confusing. 
How and why?

Reply:
i rephrased